### DISCUSSION TITLE
Global config for API routes

### DISCUSSION THREAD

**Discussion by mxstbr** [vercel/next.js#21006] [Ideas]
I'd like a global (i.e. in `next.config.js`) option to set `externalResolver: true` for all my API routes. Since there's likely a use case for globally changing the default API route config, I reckon the best solution for this might be to add the `api` config key to `next.config.js` like so:

```JS
// next.config.js

module.exports = {
  api: {
    externalResolver: true
  }
}
```

I want to set this _globally_ as I'm building an app with authentication and I'm using passport for the authentication. Because I don't care about the `next-connect` API, I instead added a `withPassport` higher-order function inspired by https://github.com/andycmaj/nextjs-passport-session-auth/blob/master/lib/withPassport.ts.

The issue is that every single one of my API routes is wrapped in `withPassport`, and thus every single one of them triggers the `externalResolver` "warning" as it's using an external resolver. This is, quite frankly, very annoying as it's a very noisy warning that doesn't even help me in this case.

I can turn that off by adding the `export const config` to every. single. API route, but that's super tedious. Hence why I'd like to add the ability to globally set the API routes config!

**Edit: first PR to add this is here https://github.com/vercel/next.js/pull/21007**

**Comment by josiahwiebe** [comment]
Would love to see this as well!
